plugin/pkg/shoot/quotavalidator/admission.go

```diff
@@ -325,7 +326,12 @@ func (q *QuotaValidator) findShootsReferQuota(quota core.Quota, shoot core.Shoot
 				continue
 			}
 			if ptr.Deref(s.Spec.SecretBindingName, "") == binding.Name {
-				shootsReferQuota = append(shootsReferQuota, *s)
+				coreShoot := &core.Shoot{}
+				if err := gardencorev1beta1.Convert_v1beta1_Shoot_To_core_Shoot(s, coreShoot, nil); err != nil {
```

Comment:
Do we need this conversion? This is the shoot plugin, yes, the shoot is in internal version, but this shoot list is used to compare with the cloudprofile which is versioned, so isn't it better to usse v1beta1 shoots from the lister directly? WDYT?

Reply:
Atleast at one place we will have to make the conversion.
Removing the conversion here, we will have to change the return type of this function to `gardencorev1beta1.shoot` and so on.

Reply:
okay, let's keep it then